html/includes/authenticate.inc.php

```diff
@@ -56,9 +56,13 @@
                 $username = clean($_REQUEST['username']);
                 $password = $_REQUEST['password'];
             } elseif (isset($_SERVER['REMOTE_USER'])) {
-                $username = $_SERVER['REMOTE_USER'];
-            } elseif (isset($_SERVER['PHP_AUTH_USER']) && $config['auth_mechanism'] === 'http-auth') {
-                $username = $_SERVER['PHP_AUTH_USER'];
+                $username = clean($_SERVER['REMOTE_USER']);
+            } elseif ($config['auth_mechanism'] === 'http-auth') {
+                if (isset($_SERVER['PHP_AUTH_USER']) && $_SERVER['PHP_AUTH_USER']) {
+                    $username = clean($_SERVER['PHP_AUTH_USER']);
+                } elseif (isset($config['http_auth_guest'])) {
```

Comment:
Yeah, I wasn't sure and honestly it doesn't matter.  Inside the http-auth class we check things properly.

I just wanted to be sure that $username is set because of the `isset($username)` before authenticate().

If you are confident PHP_AUTH_USER will be set 100% we can drop the entire elseif.  Perhaps in fact if PHP_AUTH_USER is not set we shouldn't allow guest?